modules/core/src/utils/samples.cpp

```diff
@@ -49,6 +49,7 @@ CV_EXPORTS void addSamplesDataSearchSubDirectory(const cv::String& subdir)
 
 cv::String findFile(const cv::String& relative_path, bool required, bool silentMode)
 {
+#if OPENCV_HAVE_FILESYSTEM_SUPPORT
```

Comment:
Please add include `opencv2/core/utils/filesystem.private.hpp` into this file.

(BTW, we can see how `Wundef` work on Linux builders for this issue on OpenCV CI)

Reply:
Done! I have also included the same header file in the core tests that use filesystem feature. I had to do so to disable such tests, as they were failing: https://pullrequest.opencv.org/buildbot/builders/precommit_linux64/builds/31476/steps/test_core/logs/tests%20summary

I disabled both tests even if the second one was not failing because it made sense to do so, as they both require filesystem support.